optimum/intel/openvino/modeling_text2speech.py

```diff
@@ -0,0 +1,678 @@
+                `transformers.file_utils.default_cache_path`, which is the cache directory for transformers.
+            token (Optional[Union[bool, str]], defaults to `None`):
+                The token to use as HTTP bearer authorization for remote files. If `True`, will use the token generated
+                when running `huggingface-cli login` (stored in `~/.huggingface`).
+            revision (`str`):
+                Revision is the specific model version to use. It can be a branch name, a tag name, or a commit id
+            kwargs (`Dict`, *optional*):
+                kwargs will be passed to the model during initialization
+        """
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        # This attribute is needed to keep one reference on the temporary directory, since garbage collecting
+        # would end-up removing the directory containing the underlying OpenVINO model
+        cls._model_save_dir_tempdirectory_instance = save_dir
+
+        if task is None:
+            task = cls.export_feature
+            if use_cache:
+                task = task + "-with-past"
+
+        compile_only = kwargs.pop("compile_only", False)
+        if compile_only:
+            logger.warning(
+                "`compile_only` mode will be disabled because it does not support model export."
+                "Please provide openvino model obtained using optimum-cli or saved on disk using `save_pretrained`"
+            )
+            compile_only = False
+        # If load_in_8bit and quantization_config not specified then ov_config is set to None and will be set by default in convert depending on the model size
+        if load_in_8bit is None and not quantization_config:
+            ov_config = None
+        else:
+            ov_config = OVConfig(dtype="fp32")
+        stateful = kwargs.get("stateful", True)
+        variant = kwargs.pop("variant", None)
+
+        # NOTE: need to pass kwargs that can contain vocoder path
+        # it will be an additional component in the pipeline
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            task=task,
+            subfolder=subfolder,
+            revision=revision,
+            cache_dir=cache_dir,
+            token=token,
+            local_files_only=local_files_only,
+            force_download=force_download,
+            trust_remote_code=trust_remote_code,
+            ov_config=ov_config,
+            stateful=stateful,
+            variant=variant,
+            model_kwargs=kwargs,
+        )
+
+        return cls._from_pretrained(
+            model_id=save_dir_path,
+            config=config,
+            use_cache=use_cache,
+            load_in_8bit=load_in_8bit,
+            quantization_config=quantization_config,
+            compile_only=compile_only,
+            **kwargs,
+        )
+
+    def generate(
```

Comment:
could you add a link to the original code ? https://github.com/huggingface/transformers/blob/af6d2756d9ba7d83388f617b612009242ca46681/src/transformers/models/speecht5/modeling_speecht5.py#L2427

also isn't it specific to SpeechT5 in which case we might want to move this part from OVModelForTextToSpeechSeq2Seq to something more specific to speecht5

Reply:
Also can't we use `SpeechT5ForTextToSpeech.generate` directly, the same way we do for Whisper for example, we don't need to rewrite the whole logic if our submodels are compliant and behave similarly to the transformers submodels. We also don't wanna port every fix or change from transformers to optimum-intel manually.

Reply:
@echarlaix, added a link to the original `generate` method.
@IlyasMoutawwakil, unfortunately, it is not possible to re-use original `generate` method because we combined several decoder parts (prenet, wrapper_decoder, and feat_out you can see in the original code) used in `generate` method into the single piece decoder. We did this to have 4 IRs (encoder, decoder, postnet and vocoder) instead of 4 + 2 IRs. It will be more efficient in terms of performance, etc. I would rather lean to refactor SpeechT5 code to align.